lib/ansible/modules/cloud/amazon/rds.py

```diff
@@ -1352,6 +1360,7 @@ def main():
 
     module = AnsibleModule(
         argument_spec=argument_spec,
+        supports_check_mode=True,
```

Comment:
Currently RDS is in the transition of being split up into smaller modules and ported to boto3 so eventually this can be deprecated. Here is the PR for rds_instance https://github.com/ansible/ansible/pull/26602. Supporting check_mode hasn't been added to that either yet. Since you're working with this module having your feedback on it would be great. :)